rustup-init.ps1

```diff
@@ -0,0 +1,174 @@
+        "64-bit" {
+            $cputype = "x86_64"
+            break
+        }
+        # Reported by Darin/Linux 64-bit operating systems
+        "x86_64" {
+            break
+        }
+        default {
+            Write-Error "unknown CPU type: $cputype"
+        }
+    }
+
+    $arch = "${cputype}-${ostype}"
+    Write-Debug "Detected arch '$arch'"
+    return "$arch"
+}
+
+# If RUSTUP_UPDATE_ROOT is unset, default it.
+function Get-RustupUpdateRoot {
+    if (Test-Path env:\RUSTUP_UPDATE_ROOT) {
+        return "$env:RUSTUP_UPDATE_ROOT"
+    } else {
+        return "https://static.rust-lang.org/rustup"
+    }
+}
+
+function Install-Rustup([AllowEmptyCollection()] [string[]]$Argv) {
+    $arch = Get-Architecture
+    switch -Wildcard ($arch) {
+        "*windows*" { $ext = ".exe"; break }
+        default { $ext = "" }
+    }
+
+    $url = "$(Get-RustupUpdateRoot)/dist/$arch/rustup-init$ext"
+
+    $dir = New-TemporaryDirectory
+    $file = Join-Path "$dir" "rustup-init$ext"
+
+    try {
+        if ("$arch" -match "windows") { Set-HighestEncryption }
+
+        Write-Debug "Downloading $url to $file"
+        (New-Object System.Net.WebClient).DownloadFile($url, $file)
+        if (-Not ("$arch" -match "windows")) { chmod u+x "$file" }
+
+        Write-Debug "Calling '$file $Argv'"
+        & "$file" @Argv
+    } finally {
+        Write-Debug "Cleaning $dir"
+        Remove-Item "$dir" -Force -Recurse
+    }
+}
+
+function New-TemporaryDirectory {
+    $parent = [System.IO.Path]::GetTempPath()
+    [string]$name = [System.Guid]::NewGuid()
+    New-Item -ItemType Directory -Path (Join-Path "$parent" "$name")
+}
+
+function Set-HighestEncryption {
+    # Implementation adapted from the Chocolatey installer script,
+    # see: https://chocolatey.org/install.ps1
+
+    # Attempt to set highest encryption available for SecurityProtocol.
+    # PowerShell will not set this by default (until maybe .NET 4.6.x). This
+    # will typically produce a message for PowerShell v2 (just an info message
+    # though)
+    try {
+        # Set TLS 1.2 (3072), then TLS 1.1 (768), then TLS 1.0 (192), finally
+        # SSL 3.0 (48) Use integers because the enumeration values for TLS 1.2
+        # and TLS 1.1 won't exist in .NET 4.0, even though they are addressable
+        # if .NET 4.5+ is installed (.NET 4.5 is an in-place upgrade).
+        [System.Net.ServicePointManager]::SecurityProtocol = 3072 -bor 768 -bor 192 -bor 48
+    } catch {
+        Write-Warning @'
+rustup: Unable to set PowerShell to use TLS 1.2 and TLS 1.1 due to old .NET
+Framework installed. If you see underlying connection closed or trust errors,
+you may need to do one or more of the following: (1) upgrade to .NET Framework
+4.5+ and PowerShell v3, (2) use the Download of rustup-init.exe method of
+install. See https://rustup.rs/ for all install options.
+'@
+    }
+}
+
+function Write-Error([string]$Message) {
+    Write-Warning "rustup: $Message"
+    throw
+}
+
+main $args
```

Comment:
`main` is not a recommended function name.  Functions should follow the verb-noun syntax.  Suggest 'Invoke-Main`

Reply:
You can also add a parameter block with cmdlet binding to the script itself (as in @glennsarti's [comment above](https://github.com/rust-lang/rustup.rs/pull/1878/files#r287902297) which essentially turns the script into a function which gets you all of the built-in PowerShell parameter binding, checking, and help. This basically makes the script operates like a  function and the logic you currently have in `Install-Rustup` should be in the body of the script and you can remove `main`.

i.e.
```powershell
Get-Help rustup-init.ps1
```

Reply:
I'll comment below about the main/params/etc